api/src/db/db.service.ts

```diff
@@ -323,27 +324,43 @@ export class DbService extends EventEmitter {
      * @returns - Promise containing the query result
      */
     getDocsPerGroup(userId: string, options: GetDocsOptions): Promise<DbQueryResult> {
-        // Set default options
-        if (!options.from) options.from = 0;
-
         return new Promise((resolve, reject) => {
             // To allow effective indexing, the structure inside an "$or" selector should be identical for all the sub-selectors
             // within the "$or". Because of this restriction, it is necessary to do multiple queries and join the result externally
 
             const pList = [];
 
-            const timeSelector = {
-                updatedTimeUtc: {
-                    $gte: options.from,
-                },
-            };
+            const timeSelector = [];
+            let selectors = timeSelector;
+            if (options.from && options.to) {
+                timeSelector.push({
+                    $and: [],
+                });
+                selectors = timeSelector[0].$and;
+            }
+
+            if (options.from) {
+                selectors.push({
+                    updatedTimeUtc: {
+                        $gte: options.from - this.syncTolerance,
+                    },
+                });
+            }
+
+            if (options.to) {
+                selectors.push({
+                    updatedTimeUtc: {
+                        $lt: options.to + this.syncTolerance,
+                    },
+                });
+            }
```

Comment:
I'm a bit confused by this. `selectors` does not seem to be used, does it work by reference? Why is it needed to have a separate variable for it?

Reply:
Simplified this a bit to make it more readable